view/rooms/app.js

```diff
@@ -1009,6 +1011,50 @@ function setupColumns() {
   });
 }
 
+/**
+ * Infers roomType from url parameters other than roomType
+ * (used when it is not specified)
+ * @param {object} urlParams - JSON object of url params
+ * @returns {string} the first roomType name matching one of the url params
+ */
+function getRoomTypeFromMapping(urlParams) {
+  if (!_defaultRoomType) {
+    return '';
+  }
+  const urlParamNames = Object.keys(urlParams);
+  for (let i = 0; i < urlParamNames.length; i++){
+    const paramName = urlParamNames[i];
+    const urlParamValue = urlParams[paramName];
+    const newRoomType = urlParamValue && _roomTypeMapping[paramName] &&
```

Comment:
is this returning a string or a boolean?

Reply:
A string - if it exists at the index